File: teledonCS/model/Donation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace teledonCS.model
{
    class Donation
    {
        public Donation(int id, Donor donor, CharitableCase charitableCase, double amountDonated)
        {
            this.id = id;
            this.donor = donor;
            this.charitableCase = charitableCase;
            this.amountDonated = amountDonated;
        }

        public int id { get; set; }
        public Donor donor { get; set; }
        public CharitableCase charitableCase { get; set; }
        public double amountDonated { get; set; }

        public override string ToString()
        {
            return donor.ToString() + " " + charitableCase.ToString() + " " + amountDonated.ToString();
        }
    }
}

File: TeledonSockets/TeledonModel/src/main/java/model/Donation.java
package model;


import java.io.Serializable;

public class Donation implements Identifiable<Integer>, Serializable {
    private int id;
    private Donor donor;
    private CharitableCase charitableCase;
    private int amountDonated;

    public Donation( Donor donorId, CharitableCase charitableCaseId, int amountDonated) {
        this.donor = donorId;
        this.charitableCase = charitableCaseId;
        this.amountDonated = amountDonated;
    }

    @Override
    public Integer getId() {
        return id;
    }
    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public Donor getDonor() {
        return donor;
    }

    public void setDonor(Donor donorId) {
        this.donor = donorId;
    }

    public CharitableCase getCharitableCase() {
        return charitableCase;
    }

    public void setCharitableCase(CharitableCase charitableCaseId) {
        this.charitableCase = charitableCaseId;
    }

    public int getAmountDonated() {
        return amountDonated;
    }

    public void setAmountDonated(int amountDonated) {
        this.amountDonated = amountDonated;
    }

    @Override
    public String toString() {
        return "Donation{" +
                "id=" + id +
                ", donorId=" + donor +
                ", charitableCaseId=" + charitableCase +
                ", amountDonated=" + amountDonated +
                '}';
    }
}
